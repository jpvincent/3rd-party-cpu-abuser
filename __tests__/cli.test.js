const cpuAbuser = require('../index')
const mockPath = './__tests__/__mocks__/trace-vanity-mobile.json'
const cmd = require('node-cmd')


describe('Cli executable output', () => {
	jest.setTimeout(20000)

  it('default options snapshot test', () => {
    // http://riaevangelist.github.io/node-cmd/
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath}`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

  it('options -t=0 snapshot test', () => {
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath} -t=0`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

  it('options --min-time=0 snapshot test', () => {
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath} --min-time=0`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

  it('options --json snapshot test', () => {
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath} --json`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

  it('options --groupBy EventName snapshot test', () => {
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath} --groupBy EventName`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

  it('options --endMark firstPaint snapshot test', () => {
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath} --endMark firstPaint`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

  it('options --startMark loadEventEnd --min-time=50 snapshot test', () => {
    return new Promise( (resolve, reject) => {
      cmd.get(`node cli.js ${mockPath} --startMark loadEventEnd --min-time=50`, (err, output, stderr) => {
        if(err) return reject(stderr)
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })

})
