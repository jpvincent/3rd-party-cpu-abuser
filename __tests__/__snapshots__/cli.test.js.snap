// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cli executable output default options snapshot test 1`] = `
"Analyzing ./__tests__/__mocks__/trace-vanity-mobile.json
┌──────────────────────────┬─────────┐
│ Total CPU busy time (ms) │ 7189.36 │
├──────────────────────────┼─────────┤
│ Total number of domains  │ 55      │
├──────────────────────────┼─────────┤
│ Number of big offenders  │ 14      │
└──────────────────────────┴─────────┘
┌───────────────┬────────────────────────────────┐
│ CPU Time (ms) │ domain name                    │
├───────────────┼────────────────────────────────┤
│ 1062.52       │ m.vanityfair.fr                │
├───────────────┼────────────────────────────────┤
│ 801.21        │ securepubads.g.doubleclick.net │
├───────────────┼────────────────────────────────┤
│ 601.33        │ cdns.gigya.com                 │
├───────────────┼────────────────────────────────┤
│ 560.26        │ fw.adsafeprotected.com         │
├───────────────┼────────────────────────────────┤
│ 437.32        │ z.moatads.com                  │
├───────────────┼────────────────────────────────┤
│ 427.44        │ widgets.outbrain.com           │
├───────────────┼────────────────────────────────┤
│ 411.74        │ connect.facebook.net           │
├───────────────┼────────────────────────────────┤
│ 396.08        │ pcotag.azureedge.net           │
├───────────────┼────────────────────────────────┤
│ 347.20        │ adyoulike.omnitagjs.com        │
├───────────────┼────────────────────────────────┤
│ 267.32        │ platform.twitter.com           │
├───────────────┼────────────────────────────────┤
│ 224.96        │ cdns.eu1.gigya.com             │
├───────────────┼────────────────────────────────┤
│ 207.51        │ secure-ds.serving-sys.com      │
├───────────────┼────────────────────────────────┤
│ 192.87        │ www.googletagmanager.com       │
├───────────────┼────────────────────────────────┤
│ 169.77        │ pagead2.googlesyndication.com  │
└───────────────┴────────────────────────────────┘
"
`;

exports[`Cli executable output options --endMark firstPaint snapshot test 1`] = `
"Analyzing ./__tests__/__mocks__/trace-vanity-mobile.json
┌──────────────────────────┬────────┐
│ Total CPU busy time (ms) │ 239.39 │
├──────────────────────────┼────────┤
│ Total number of domains  │ 12     │
├──────────────────────────┼────────┤
│ Number of big offenders  │ 0      │
└──────────────────────────┴────────┘
┌───────────────┬─────────────┐
│ CPU Time (ms) │ domain name │
└───────────────┴─────────────┘
"
`;

exports[`Cli executable output options --groupBy EventName snapshot test 1`] = `
"Analyzing ./__tests__/__mocks__/trace-vanity-mobile.json
┌──────────────────────────┬─────────┐
│ Total CPU busy time (ms) │ 8148.96 │
├──────────────────────────┼─────────┤
│ Total number of domains  │ 17      │
├──────────────────────────┼─────────┤
│ Number of big offenders  │ 6       │
└──────────────────────────┴─────────┘
┌───────────────┬───────────────────┐
│ CPU Time (ms) │ domain name       │
├───────────────┼───────────────────┤
│ 5550.70       │ JS Frame          │
├───────────────┼───────────────────┤
│ 960.97        │ Compile Script    │
├───────────────┼───────────────────┤
│ 461.06        │ Recalculate Style │
├───────────────┼───────────────────┤
│ 290.16        │ Layout            │
├───────────────┼───────────────────┤
│ 222.63        │ Parse HTML        │
├───────────────┼───────────────────┤
│ 200.95        │ Update Layer Tree │
└───────────────┴───────────────────┘
"
`;

exports[`Cli executable output options --json snapshot test 1`] = `
"{ totalCPUTime: 7189.355000168085,
  totalDomains: 55,
  totalOffendersDomains: 14,
  cpuTimePerDomain: 
   { 'm.vanityfair.fr': 1062.517999947071,
     'securepubads.g.doubleclick.net': 801.206000238657,
     'cdns.gigya.com': 601.3290000259876,
     'fw.adsafeprotected.com': 560.256000071764,
     'z.moatads.com': 437.3180001080036,
     'widgets.outbrain.com': 427.435999751091,
     'connect.facebook.net': 411.74199998378754,
     'pcotag.azureedge.net': 396.0799999833107,
     'adyoulike.omnitagjs.com': 347.20399993658066,
     'platform.twitter.com': 267.3179999291897,
     'cdns.eu1.gigya.com': 224.9640000462532,
     'secure-ds.serving-sys.com': 207.50500014424324,
     'www.googletagmanager.com': 192.87200009822845,
     'pagead2.googlesyndication.com': 169.76800015568733 } }
"
`;

exports[`Cli executable output options --min-time=0 snapshot test 1`] = `
"Analyzing ./__tests__/__mocks__/trace-vanity-mobile.json
┌──────────────────────────┬─────────┐
│ Total CPU busy time (ms) │ 7189.36 │
├──────────────────────────┼─────────┤
│ Total number of domains  │ 55      │
├──────────────────────────┼─────────┤
│ Number of big offenders  │ 55      │
└──────────────────────────┴─────────┘
┌───────────────┬─────────────────────────────────┐
│ CPU Time (ms) │ domain name                     │
├───────────────┼─────────────────────────────────┤
│ 1062.52       │ m.vanityfair.fr                 │
├───────────────┼─────────────────────────────────┤
│ 801.21        │ securepubads.g.doubleclick.net  │
├───────────────┼─────────────────────────────────┤
│ 601.33        │ cdns.gigya.com                  │
├───────────────┼─────────────────────────────────┤
│ 560.26        │ fw.adsafeprotected.com          │
├───────────────┼─────────────────────────────────┤
│ 437.32        │ z.moatads.com                   │
├───────────────┼─────────────────────────────────┤
│ 427.44        │ widgets.outbrain.com            │
├───────────────┼─────────────────────────────────┤
│ 411.74        │ connect.facebook.net            │
├───────────────┼─────────────────────────────────┤
│ 396.08        │ pcotag.azureedge.net            │
├───────────────┼─────────────────────────────────┤
│ 347.20        │ adyoulike.omnitagjs.com         │
├───────────────┼─────────────────────────────────┤
│ 267.32        │ platform.twitter.com            │
├───────────────┼─────────────────────────────────┤
│ 224.96        │ cdns.eu1.gigya.com              │
├───────────────┼─────────────────────────────────┤
│ 207.51        │ secure-ds.serving-sys.com       │
├───────────────┼─────────────────────────────────┤
│ 192.87        │ www.googletagmanager.com        │
├───────────────┼─────────────────────────────────┤
│ 169.77        │ pagead2.googlesyndication.com   │
├───────────────┼─────────────────────────────────┤
│ 143.86        │ tpc.googlesyndication.com       │
├───────────────┼─────────────────────────────────┤
│ 112.25        │ cdns1.gigya.com                 │
├───────────────┼─────────────────────────────────┤
│ 109.66        │ libs.sphere.com                 │
├───────────────┼─────────────────────────────────┤
│ 89.03         │ tags.bkrtx.com                  │
├───────────────┼─────────────────────────────────┤
│ 80.26         │ www.google-analytics.com        │
├───────────────┼─────────────────────────────────┤
│ 67.68         │ assets.pinterest.com            │
├───────────────┼─────────────────────────────────┤
│ 60.24         │ www.googletagservices.com       │
├───────────────┼─────────────────────────────────┤
│ 46.84         │ fo-static.omnitagjs.com         │
├───────────────┼─────────────────────────────────┤
│ 46.83         │ staticxx.facebook.com           │
├───────────────┼─────────────────────────────────┤
│ 44.55         │ rva.outbrain.com                │
├───────────────┼─────────────────────────────────┤
│ 43.39         │ native                          │
├───────────────┼─────────────────────────────────┤
│ 39.69         │ www.gstatic.com                 │
├───────────────┼─────────────────────────────────┤
│ 34.70         │ static.mediarithmics.com        │
├───────────────┼─────────────────────────────────┤
│ 26.05         │ ads.rubiconproject.com          │
├───────────────┼─────────────────────────────────┤
│ 19.87         │ www.instagram.com               │
├───────────────┼─────────────────────────────────┤
│ 16.73         │ www.wysistat.com                │
├───────────────┼─────────────────────────────────┤
│ 15.19         │ bs.serving-sys.com              │
├───────────────┼─────────────────────────────────┤
│ 11.46         │ stags.bluekai.com               │
├───────────────┼─────────────────────────────────┤
│ 9.77          │ tag.audience.acpm.fr            │
├───────────────┼─────────────────────────────────┤
│ 8.79          │ vastcdn.outbrain.com            │
├───────────────┼─────────────────────────────────┤
│ 8.46          │ sb.scorecardresearch.com        │
├───────────────┼─────────────────────────────────┤
│ 8.27          │ script.ownpage.fr               │
├───────────────┼─────────────────────────────────┤
│ 6.20          │ fo-ssp.omnitagjs.com            │
├───────────────┼─────────────────────────────────┤
│ 5.52          │ optimized-by.rubiconproject.com │
├───────────────┼─────────────────────────────────┤
│ 4.49          │ odb.outbrain.com                │
├───────────────┼─────────────────────────────────┤
│ 3.80          │ api.premiumcollect.com          │
├───────────────┼─────────────────────────────────┤
│ 3.70          │ www.google.com                  │
├───────────────┼─────────────────────────────────┤
│ 2.76          │ a.teads.tv                      │
├───────────────┼─────────────────────────────────┤
│ 2.20          │ storage.premiumcollect.com      │
├───────────────┼─────────────────────────────────┤
│ 1.39          │ fo-api.omnitagjs.com            │
├───────────────┼─────────────────────────────────┤
│ 1.35          │ gscounters.eu1.gigya.com        │
├───────────────┼─────────────────────────────────┤
│ 1.35          │ adservice.google.fr             │
├───────────────┼─────────────────────────────────┤
│ 1.15          │ graph.facebook.com              │
├───────────────┼─────────────────────────────────┤
│ 0.97          │ static.adsafeprotected.com      │
├───────────────┼─────────────────────────────────┤
│ 0.74          │ images.outbrainimg.com          │
├───────────────┼─────────────────────────────────┤
│ 0.69          │ revee.outbrain.com              │
├───────────────┼─────────────────────────────────┤
│ 0.53          │ adservice.google.com            │
├───────────────┼─────────────────────────────────┤
│ 0.50          │ triggermkg.leadplace.fr         │
├───────────────┼─────────────────────────────────┤
│ 0.47          │ log.outbrainimg.com             │
├───────────────┼─────────────────────────────────┤
│ 0.28          │ log.pinterest.com               │
├───────────────┼─────────────────────────────────┤
│ 0.17          │ eus.rubiconproject.com          │
└───────────────┴─────────────────────────────────┘
"
`;

exports[`Cli executable output options --startMark loadEventEnd --min-time=50 snapshot test 1`] = `
"Analyzing ./__tests__/__mocks__/trace-vanity-mobile.json
┌──────────────────────────┬─────────┐
│ Total CPU busy time (ms) │ 2312.64 │
├──────────────────────────┼─────────┤
│ Total number of domains  │ 36      │
├──────────────────────────┼─────────┤
│ Number of big offenders  │ 11      │
└──────────────────────────┴─────────┘
┌───────────────┬────────────────────────────────┐
│ CPU Time (ms) │ domain name                    │
├───────────────┼────────────────────────────────┤
│ 546.78        │ fw.adsafeprotected.com         │
├───────────────┼────────────────────────────────┤
│ 437.32        │ z.moatads.com                  │
├───────────────┼────────────────────────────────┤
│ 234.77        │ securepubads.g.doubleclick.net │
├───────────────┼────────────────────────────────┤
│ 170.16        │ secure-ds.serving-sys.com      │
├───────────────┼────────────────────────────────┤
│ 148.60        │ connect.facebook.net           │
├───────────────┼────────────────────────────────┤
│ 137.96        │ pagead2.googlesyndication.com  │
├───────────────┼────────────────────────────────┤
│ 111.17        │ tpc.googlesyndication.com      │
├───────────────┼────────────────────────────────┤
│ 88.10         │ m.vanityfair.fr                │
├───────────────┼────────────────────────────────┤
│ 76.77         │ adyoulike.omnitagjs.com        │
├───────────────┼────────────────────────────────┤
│ 71.55         │ cdns.gigya.com                 │
├───────────────┼────────────────────────────────┤
│ 54.24         │ libs.sphere.com                │
└───────────────┴────────────────────────────────┘
"
`;

exports[`Cli executable output options -t=0 snapshot test 1`] = `
"Analyzing ./__tests__/__mocks__/trace-vanity-mobile.json
┌──────────────────────────┬─────────┐
│ Total CPU busy time (ms) │ 7189.36 │
├──────────────────────────┼─────────┤
│ Total number of domains  │ 55      │
├──────────────────────────┼─────────┤
│ Number of big offenders  │ 55      │
└──────────────────────────┴─────────┘
┌───────────────┬─────────────────────────────────┐
│ CPU Time (ms) │ domain name                     │
├───────────────┼─────────────────────────────────┤
│ 1062.52       │ m.vanityfair.fr                 │
├───────────────┼─────────────────────────────────┤
│ 801.21        │ securepubads.g.doubleclick.net  │
├───────────────┼─────────────────────────────────┤
│ 601.33        │ cdns.gigya.com                  │
├───────────────┼─────────────────────────────────┤
│ 560.26        │ fw.adsafeprotected.com          │
├───────────────┼─────────────────────────────────┤
│ 437.32        │ z.moatads.com                   │
├───────────────┼─────────────────────────────────┤
│ 427.44        │ widgets.outbrain.com            │
├───────────────┼─────────────────────────────────┤
│ 411.74        │ connect.facebook.net            │
├───────────────┼─────────────────────────────────┤
│ 396.08        │ pcotag.azureedge.net            │
├───────────────┼─────────────────────────────────┤
│ 347.20        │ adyoulike.omnitagjs.com         │
├───────────────┼─────────────────────────────────┤
│ 267.32        │ platform.twitter.com            │
├───────────────┼─────────────────────────────────┤
│ 224.96        │ cdns.eu1.gigya.com              │
├───────────────┼─────────────────────────────────┤
│ 207.51        │ secure-ds.serving-sys.com       │
├───────────────┼─────────────────────────────────┤
│ 192.87        │ www.googletagmanager.com        │
├───────────────┼─────────────────────────────────┤
│ 169.77        │ pagead2.googlesyndication.com   │
├───────────────┼─────────────────────────────────┤
│ 143.86        │ tpc.googlesyndication.com       │
├───────────────┼─────────────────────────────────┤
│ 112.25        │ cdns1.gigya.com                 │
├───────────────┼─────────────────────────────────┤
│ 109.66        │ libs.sphere.com                 │
├───────────────┼─────────────────────────────────┤
│ 89.03         │ tags.bkrtx.com                  │
├───────────────┼─────────────────────────────────┤
│ 80.26         │ www.google-analytics.com        │
├───────────────┼─────────────────────────────────┤
│ 67.68         │ assets.pinterest.com            │
├───────────────┼─────────────────────────────────┤
│ 60.24         │ www.googletagservices.com       │
├───────────────┼─────────────────────────────────┤
│ 46.84         │ fo-static.omnitagjs.com         │
├───────────────┼─────────────────────────────────┤
│ 46.83         │ staticxx.facebook.com           │
├───────────────┼─────────────────────────────────┤
│ 44.55         │ rva.outbrain.com                │
├───────────────┼─────────────────────────────────┤
│ 43.39         │ native                          │
├───────────────┼─────────────────────────────────┤
│ 39.69         │ www.gstatic.com                 │
├───────────────┼─────────────────────────────────┤
│ 34.70         │ static.mediarithmics.com        │
├───────────────┼─────────────────────────────────┤
│ 26.05         │ ads.rubiconproject.com          │
├───────────────┼─────────────────────────────────┤
│ 19.87         │ www.instagram.com               │
├───────────────┼─────────────────────────────────┤
│ 16.73         │ www.wysistat.com                │
├───────────────┼─────────────────────────────────┤
│ 15.19         │ bs.serving-sys.com              │
├───────────────┼─────────────────────────────────┤
│ 11.46         │ stags.bluekai.com               │
├───────────────┼─────────────────────────────────┤
│ 9.77          │ tag.audience.acpm.fr            │
├───────────────┼─────────────────────────────────┤
│ 8.79          │ vastcdn.outbrain.com            │
├───────────────┼─────────────────────────────────┤
│ 8.46          │ sb.scorecardresearch.com        │
├───────────────┼─────────────────────────────────┤
│ 8.27          │ script.ownpage.fr               │
├───────────────┼─────────────────────────────────┤
│ 6.20          │ fo-ssp.omnitagjs.com            │
├───────────────┼─────────────────────────────────┤
│ 5.52          │ optimized-by.rubiconproject.com │
├───────────────┼─────────────────────────────────┤
│ 4.49          │ odb.outbrain.com                │
├───────────────┼─────────────────────────────────┤
│ 3.80          │ api.premiumcollect.com          │
├───────────────┼─────────────────────────────────┤
│ 3.70          │ www.google.com                  │
├───────────────┼─────────────────────────────────┤
│ 2.76          │ a.teads.tv                      │
├───────────────┼─────────────────────────────────┤
│ 2.20          │ storage.premiumcollect.com      │
├───────────────┼─────────────────────────────────┤
│ 1.39          │ fo-api.omnitagjs.com            │
├───────────────┼─────────────────────────────────┤
│ 1.35          │ gscounters.eu1.gigya.com        │
├───────────────┼─────────────────────────────────┤
│ 1.35          │ adservice.google.fr             │
├───────────────┼─────────────────────────────────┤
│ 1.15          │ graph.facebook.com              │
├───────────────┼─────────────────────────────────┤
│ 0.97          │ static.adsafeprotected.com      │
├───────────────┼─────────────────────────────────┤
│ 0.74          │ images.outbrainimg.com          │
├───────────────┼─────────────────────────────────┤
│ 0.69          │ revee.outbrain.com              │
├───────────────┼─────────────────────────────────┤
│ 0.53          │ adservice.google.com            │
├───────────────┼─────────────────────────────────┤
│ 0.50          │ triggermkg.leadplace.fr         │
├───────────────┼─────────────────────────────────┤
│ 0.47          │ log.outbrainimg.com             │
├───────────────┼─────────────────────────────────┤
│ 0.28          │ log.pinterest.com               │
├───────────────┼─────────────────────────────────┤
│ 0.17          │ eus.rubiconproject.com          │
└───────────────┴─────────────────────────────────┘
"
`;
